test/e2e/storage/framework/group_snapshot_resource.go

```diff
@@ -0,0 +1,126 @@
+				"source": map[string]interface{}{
+					"selector": map[string]interface{}{
+						"matchLabels": map[string]interface{}{
+							"group": groupName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return snapshot
+}
+
+// GroupSnapshotResource represents a groupsnapshot class, a groupsnapshot and its bound contents for a specific test case
+type GroupSnapshotResource struct {
+	Config  *PerTestConfig
+	Pattern TestPattern
+
+	Vgs        *unstructured.Unstructured
+	Vgscontent *unstructured.Unstructured
+	Vgsclass   *unstructured.Unstructured
+}
+
+// CreateGroupSnapshot creates a VolumeGroupSnapshotClass with given SnapshotDeletionPolicy and a VolumeGroupSnapshot
+// from the VolumeGroupSnapshotClass using a dynamic client.
+// Returns the unstructured VolumeGroupSnapshotClass and VolumeGroupSnapshot objects.
+func CreateGroupSnapshot(ctx context.Context, sDriver VoulmeGroupSnapshottableTestDriver, config *PerTestConfig, pattern TestPattern, groupName string, pvcNamespace string, timeouts *framework.TimeoutContext, parameters map[string]string) (*unstructured.Unstructured, *unstructured.Unstructured) {
+	defer ginkgo.GinkgoRecover()
+	var err error
+	if pattern.SnapshotType != GroupVolumeSnapshot {
+		err = fmt.Errorf("SnapshotType must be set to GroupVolumeSnapshot")
+		framework.ExpectNoError(err)
+	}
+	dc := config.Framework.DynamicClient
+
+	ginkgo.By("creating a GroupSnapshotClass")
+	gsclass := sDriver.GetVolumeGroupSnapshotClass(ctx, config, parameters)
+	if gsclass == nil {
+		framework.Failf("Failed to get group snapshot class based on test config")
+	}
+	gsclass.Object["deletionPolicy"] = pattern.SnapshotDeletionPolicy.String()
+
+	gsclass, err = dc.Resource(utils.VolumeGroupSnapshotClassGVR).Create(ctx, gsclass, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	gsclass, err = dc.Resource(utils.VolumeGroupSnapshotClassGVR).Get(ctx, gsclass.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("creating a dynamic GroupVolumeSnapshot")
+	// Prepare a dynamically provisioned group volume snapshot with certain data
+	groupSnapshot := getGroupSnapshot(groupName, pvcNamespace, gsclass.GetName())
+
+	groupSnapshot, err = dc.Resource(utils.VolumeGroupSnapshotGVR).Namespace(groupSnapshot.GetNamespace()).Create(ctx, groupSnapshot, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	ginkgo.By("Waiting for group snapshot to be ready")
+	err = utils.WaitForGroupSnapshotReady(ctx, dc, groupSnapshot.GetNamespace(), groupSnapshot.GetName(), framework.Poll, timeouts.SnapshotCreate)
+	framework.ExpectNoError(err)
+	ginkgo.By("Getting group snapshot and content")
+	groupSnapshot, err = dc.Resource(utils.VolumeGroupSnapshotGVR).Namespace(groupSnapshot.GetNamespace()).Get(ctx, groupSnapshot.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	return gsclass, groupSnapshot
+}
+
+// CleanupResource deletes the VolumeGroupSnapshotClass and VolumeGroupSnapshot objects using a dynamic client.
```

Comment:
we are deleting only the class not the volumeGroupSnapshot, this comment need to be updated as well.